src/provider/provider_os_memory.c

```diff
@@ -637,7 +637,8 @@ static inline void assert_is_page_aligned(uintptr_t ptr, size_t page_size) {
 static int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment,
```

Comment:
NIT: 12th argument to this function. This screams for refactor.

Reply:
Yes, I will do that in a separate PR.